daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/PackedBinaryTraits.scala

```diff
@@ -69,17 +78,35 @@ abstract class PackedBinaryDecimalBaseParser(
 }
 
 abstract class PackedBinaryIntegerBaseParser(
-  override val context: ElementRuntimeData,
-  signed: Boolean = false
+  override val context: ElementRuntimeData
 ) extends PrimParser
   with PackedBinaryConversion {
   override lazy val runtimeDependencies = Vector()
 
+  val signed = {
+    context.optPrimType.get match {
+      case n: NodeInfo.PrimType.PrimNumeric => n.isSigned
+      // context.optPrimType can be of type date/time via ConvertZonedCombinator
+      case _ => false
+    }
+  }
   protected def getBitLength(s: ParseOrUnparseState): Int
 
   def parse(start: PState): Unit = {
     val nBits = getBitLength(start)
-    if (nBits == 0) return // zero length is used for outputValueCalc often.
+    if (nBits == 0) {
+      val signedStr = if (signed) "a signed" else "an unsigned"
+      val minWidth = if (signed) 2 else 1
+      PE(
+        start,
+        "Minimum length for %s binary integer is %d bit(s), number of bits %d out of range. " +
+          "An unsigned integer with length 1 bit could be used instead.",
+        signedStr,
+        minWidth,
+        nBits
+      )
+      return
+    }
```

Comment:
Same question as above, I'm not sure this error makes sense for packed numbers and can possibly be removed?